Root data source

chait-slim opened the pull request:
Relevant tickets for reference:
https://github.com/google/osv.dev/issues/4396#issue-3662334257
https://github.com/google/osv.dev/issues/3937

chait-slim commented on source.yaml:
Done. The other PR: https://github.com/google/osv.dev/pull/4548

another-rex commented:
/gcbrun

another-rex commented:
Please have a look at the linting errors. (You can run this locally with make lint)

another-rex commented:
Also, tests seem to be failing:

```
OK
+ poetry run python -m unittest osv.purl_helpers_test
.F
======================================================================
FAIL: tests_package_to_purl (osv.purl_helpers_test.PurlHelpersTest.tests_package_to_purl)
Test PURL generation.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/osv/purl_helpers_test.py", line 140, in tests_package_to_purl
    self.assertEqual('pkg:generic/root/%40root%2Flodash',
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                     purl_helpers.package_to_purl('Root', '@root/lodash'))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 'pkg:generic/root/%40root%2Flodash' != 'pkg:generic/root/%40root/lodash'
- pkg:generic/root/%40root%2Flodash
?                         ^^^
+ pkg:generic/root/%40root/lodash
?                         ^
```

you can run it locally with the poetry run python -m unittest osv.purl_helpers_test , or with make all-tests

chait-slim commented:
@another-rex 
1. Fixed lint issues (ran make lint locally to verify)
2. Fixed failing tests (ran make all-tests locally). This required a change in the purl_helpers.py file. The issue is that that Root is first ecosystem with a namespace + slash in the PURL.

chait-slim commented:
Opened:
https://github.com/google/osv.dev/pull/4558
https://github.com/google/osv.dev/pull/4559
For the source files as requested

chait-slim commented:
@another-rex I see tests failing but not sure how to see which are failing and why, since locally make all-tests passed

jess-lowe commented:
/gcbrun
